Add header html tag to CSS for `monobackgroundcolor`

mcanouil opened the pull request:
This PR adds h1-6 tags to the CSS rules to apply `monobackgroundcolor` on.

Fixes #6781

cderv committed:
chores: Use a formatter for our R code based on REditorSupport extensions

cderv committed:
[knitr] Correctly create `tools:quarto` env to store ojs_define for knitr to find during knitting

This avoids duplication of environment, as discussed in https://github.com/quarto-dev/quarto-cli/pull/5012#pullrequestreview-1362676666

cderv committed:
[knitr] Reformat with vscode the script

cderv committed:
[knitr] handles OJS conditional with other dependency resolution

This simplifies ojs_define logic removing need for a custom function in a tools env

cderv committed:
[knitr] Rewrite dependencies handling by format and include ojs with HTML

cderv committed:
Add test to avoid duplicated tools:quarto env

mcanouil committed:
fix: `monobackgroundcolor` ignored by document title and section title

Fixes #6781

mcanouil committed:
fix: `monobackgroundcolor` ignored by document title and section title

Fixes #6781

mcanouil committed:
fix: add header to inline code style without padding

mcanouil committed:
fix: add header to inline code style without padding

mcanouil committed:
fix: `monobackgroundcolor` ignored by document title and section title

Fixes #6781

mcanouil committed:
fix: `monobackgroundcolor` ignored by document title and section title

Fixes #6781

dragonstyle committed:
Improve funding schema for metadata

- allow mutiple awards for a funding
- correct resolve references
- introduce tests that will help validate JATS output

dragonstyle committed:
Add additional license types and cleanup link text

Partially address #6783

Doesn’t use badges

mcanouil committed:
fix: Add missing CC licences and display CC code instead of URL

Fixes #6783

dragonstyle committed:
Fix basic test

mcanouil committed:
fix: missing CC declaration

dragonstyle committed:
correct schema

github-actions[bot] committed:
[CI] Update timing file (#6795)

Co-authored-by: cderv <cderv@users.noreply.github.com>

cderv commented on src/resources/formats/html/bootstrap/_bootstrap-rules.scss:
```diff
@@ -907,6 +907,12 @@ pre {
 // Maps the pandoc 'monobackgroundcolor' to bootstrap
 // Note this only targets code outside of sourceCode blocks
 @if variable-exists(mono-background-color) {
```
Looking at this, I am wondering if this part of the code (the oldest from https://github.com/mcanouil/quarto-cli/commit/59591f61a705414c398f0fd96d003132ae193722) is not unnecessary anymore. 
https://github.com/quarto-dev/quarto-cli/blob/e637c76b14e5327a4053ff0ffcaa8cab27b672dd/src/resources/formats/html/bootstrap/_bootstrap-rules.scss#L907-L924
It seems we are duplicating it with what is below 
https://github.com/quarto-dev/quarto-cli/blob/e637c76b14e5327a4053ff0ffcaa8cab27b672dd/src/resources/formats/html/bootstrap/_bootstrap-rules.scss#L926-L949

The later is more conditional to variables than the former, and it will overwrite it too. 

Possibly that was the issue for #4746 fixed by adding a duplicated conditional in 9fff5b7cc4ece31767c91ebf5752d979d6f07999

Just sharing this observation as it could be an opportunity to simplify. We could also leverage more SASS syntax to avoid writing and maintaining the expanded version of all tags and rules with `code:not(.sourceCode)` - but that is ok like this also.

mcanouil replied:
From my few tests, I did not notice something wrong with this "duplication" and assumed it was meant to fix something elsewhere.

cderv replied:
Yes I think it would be the second. I believe it is duplicate because when adding the second, the first already existed. And then it was overriding the first, which caused #4746 I believe.

mcanouil replied:
Merged the two rules into one (the second): f36fb64f11140c9a32605e476ae7f5c2e2a6113f

cderv reviewed (commented):
Thanks @mcanouil ! 

Just a note about possible simplification of our style file that we could do

cderv committed:
Merge branch 'main' into fix/knitr/ojs-define-env

cderv committed:
Add quarto tools  environment for ojs_define in changelog

cderv committed:
Revert `launch.json` change from aad283f48584964d8060ca4d82436271f20409ba

Even if this is to modify for testing, we don't want hardcode user path otherwise we'll commit each change each time for each user.

[skip ci]

cderv committed:
Fix support for `fig-asp` as a format or execute level YAML option

It still only works for knitr computation engine but can now correctly be used in YAML as documented in formats reference page

cderv committed:
[knitr improvements] OJS and dependencies resolution (#6775)

cderv committed:
Merge branch 'main' into knitr/fig-asp-global

mcanouil committed:
Merge branch 'main' into fix/issue6783

mcanouil committed:
fix: use license text as text instead or URL

mcanouil committed:
chore: quarto build-js

dragonstyle committed:
Merge pull request #6784 from mcanouil/fix/issue6783

fix: Add missing CC licences and display CC code instead of URL

cderv committed:
Merge pull request #6797 from quarto-dev/knitr/fig-asp-global

jjallaire committed:
initial work on pandoc 3.1.8

dragonstyle committed:
update templates

dragonstyle committed:
correct template

dragonstyle committed:
Merge pull request #6798 from quarto-dev/feature/pandoc-3.1.8

pandoc 3.1.8

dragonstyle committed:
Add no-prompt support

dragonstyle committed:
Add environment display to binder command

cderv committed:
Add tidy knitr option value in YAML validation

cderv committed:
quarto build-js

dragonstyle committed:
Add Binder prompt

cderv committed:
Merge pull request #6565 from quarto-dev/yaml/tidy-options

dragonstyle committed:
Add target to navbar tools

cscheid committed:
don't use imports straight into our vendor dir

cscheid committed:
don't use imports straight into our vendor dir

cderv committed:
don't use imports straight into our vendor dir

dragonstyle committed:
Improve Binder UI

cscheid committed:
new tool: find-bad-imports.ts to catch bad import statements in our codebase

dragonstyle committed:
Improved code-links schema

dragonstyle committed:
correctly merge link arrays

dragonstyle committed:
use correct code-links schema

dragonstyle committed:
add code link tests

cderv committed:
Adapt path to Deno on windows to new folder using architecture for quarto dev

This follows work on #6182 where modification of windows specific configure and run file where missing

cderv committed:
Also adapt build script

cderv committed:
Add a deno_std caching step

[skip ci]

cderv committed:
Merge pull request #6822 from quarto-dev/gha/cache-deno-deps

cderv committed:
Fix run-test.ps1 for new deno path too

dragonstyle committed:
Improve manuscript project defaults

dragonstyle committed:
Simply label

cderv committed:
Merge pull request #6820 from quarto-dev/fix/deno-dev-path

dragonstyle committed:
correct manuscript defaulting

dragonstyle committed:
Move Github Context gathering into Project Environment

dragonstyle committed:
Make environment discoverable as a part of project context

cscheid committed:
more robust cleanup of sourcemapping info in CSS (#6807)

* more robust cleanup of sourcemapping info in CSS

* changelog

cscheid committed:
temporarily disable ACM format until that's fixed

dragonstyle committed:
Remove extraneous logging

dragonstyle committed:
Move `code-links` into the HTML format

Any HTML format should be able to use and resolve code links. string values like `binder` and `repo` will be resolved for any project

dragonstyle committed:
Ensure book/website supports code links

dragonstyle committed:
only install proxy when vscode is being configured

jjallaire committed:
update to typst 0.8

cderv committed:
[GHA] Use actions/quarto-dev to configure Quarto from dev version on each GHA workflow

cderv committed:
[GHA] Use OS name in title to clarify on CI log

[skip ci]

cderv committed:
Add info messages before and after quarto-preview.js building step

cderv committed:
Merge pull request #6830 from quarto-dev/gha/add-caching-deno-2

cderv committed:
Add more verbose in run-tests scripts about python

cderv committed:
Improve python with pipenv to fix caching

cderv committed:
Merge pull request #6831 from quarto-dev/gha/other-improvements

cscheid committed:
Improve error message under YAML parse errors. Closes #6825 (#6832)

dragonstyle committed:
baseline lightbox filter

dragonstyle committed:
Baseline lightbox impl

- traverse topdown to more effectively screen sub elements
- process Image, Figure, and FloatRefTarget
- don’t automatically process inline images
- reduce filter passes

cscheid committed:
video shortcode: Handle partially-specified width, height, aspectratio attributes in HTML format (#6835)

* handle partially-specified height+width+aspectRatio in HTML videos
* smoke test
* changelog

dragonstyle committed:
ensure that outer flags are respected when using `filters`

dragonstyle committed:
General improvements

- hide behind flag
- only run walk on correct formats
- improve appearance

dragonstyle committed:
Add some baseline tests

dragonstyle committed:
fix tests

dragonstyle committed:
Properly form galleries from subfigures

mcanouil committed:
fix: Documentation is missing abstract from MSWord Options (#6840)

* fix: Documentation is missing abstract from MSWord Options
Fixes #6836
* chore: add changelog entry

dragonstyle committed:
Use proper full caption prefix

dragonstyle committed:
update tests

dragonstyle committed:
fix broken tests

dragonstyle committed:
Fix extension filtering logic

dragonstyle committed:
Properly use and test for non breaking spaces

cscheid committed:
RefTargets: Float (#6620)

See #6620 and #4944.

dragonstyle committed:
fix funding statement

dragonstyle committed:
fix figure captions

cscheid committed:
fix pandoc3 figure rendering in presence of linked figures

dragonstyle committed:
remove extraneous logging

dragonstyle committed:
add pandoc test

dragonstyle committed:
Merge pull request #6851 from quarto-dev/feature/lb2

Build in LightBox

dragonstyle committed:
remove unused imports

dragonstyle committed:
lightbox schema

dragonstyle committed:
default enable ligthbox for manuscripts

dragonstyle committed:
improve handling of table

cscheid committed:
remove print

cscheid committed:
improve code block

dragonstyle committed:
add glightbox to html dependencies

dragonstyle committed:
Don’t walk divs directly

dragonstyle committed:
check nil

cscheid committed:
Regression in Lua column layouts: forward classes correctly. (#6865)

* regression: forward classes correctly. Closes #6863
* add test

github-actions[bot] committed:
timing for new tests [auto PR] (#6869)

cderv committed:
`_quarto.ast.walk` needs to be applied on an element

Follow up on #6851 and part of CI fix #6871

cderv committed:
Fix test for lightbox JS lib inclusion

Follow up to #6251 and part of fixes for CI issues  #6871

cscheid committed:
better document echo behavior. See #6886

dragonstyle committed:
properly unroll divs

cderv committed:
[tests] Install dev versions for knitr and rmarkdown as part of tests environment configuration

cscheid committed:
Bugfix/687x (#6887)

* fix complex preamble layout in jats, asciidoc, add tests
* don't fail if parsed table has no captions.

cscheid committed:
be robust under section references

cderv committed:
Add some missing formats for document level Pandoc variable

cderv committed:
rebuild-js for new schema info

cscheid committed:
handle subfigures and subcaptions in latex more robustly. Closes #2492 (#6902)

cderv committed:
Update R dependencies and use latest renv tooling (#6896)

* Add a renv ignore files to speed up auto discovery of dependencies

* update to renv 1.0.2

* Use explicit discovery to improve startup speed and dependencies discovery

this means that any new R dep will need to be added to this DESCRIPTION file

* Set R version in renv project to the same one as used in CI

* Use Posit package manager with renv

* Remove repos override

* Update to renv 1.0.3

* Update renv lock file

* cleanup renv settings files

* unvalidate cache manually for clean state

* Add downlit we use for code-linking

* Try pak on CI to deal with system deps on Linux

* pak is buggy with renv resolution so install manually

* Ignore r-universe urls

* don't use renv install for installing for r-universe

cscheid committed:
Bugfix/code cell location (#6905)

* hoist cell code out of floatreftargets
* preserve DOM of 1.3 in cells
* preserve location of figure inside div
* stand back on layout cells

cderv committed:
Add about renv explicit dependencies discovery in tests README

[skip ci]

cscheid committed:
changelog (#6911)

cscheid committed:
don't crash in empty layout output (#6914)

cscheid committed:
typst: fix pandoc.Figure AST rendering

sijow committed:
deb postrm: remove pandoc only if symlink

Fixes #6901

dragonstyle committed:
Merge pull request #6921 from knuesel/fix-downgrade-pandoc

deb postrm: remove pandoc only if symlink

cderv committed:
JS file for reveal-plugin extension needs to be a EJS template

dragonstyle committed:
Display title block for HTML when other (non-title/author/subtitle) metadata is present.

Fixes #6714

dragonstyle committed:
Ensure text for external href in navigation items

If there is no text displayed for external links in navigation items (navbar, sidebar), display the href value as the text.

Fixes #6706

cderv committed:
Reconfiguration does not work on windows

Deactivate and show command to run.

It fails because `quarto.cmd` runs deno which lock the process and prevent access for removal and installation

cderv committed:
Fix configure on windows

dragonstyle committed:
Merge pull request #6925 from quarto-dev/fix/ejs-create-extension

JS file for reveal-plugin extension needs to be a EJS template

dragonstyle committed:
Don’t display play, use it

Fixes #6903

dragonstyle committed:
Error when listings don’t match any files

Fixes #5463

cderv committed:
[revealjs] Prevent last page-break in print pdf mode

Adding a CSS rule shared in #3997

dragonstyle committed:
Don’t prepent relative path to web urls

Resolves #6943

dragonstyle committed:
Within feeds, remove `index.html`

Fixes #6904

dragonstyle committed:
Update to BS Icons 1.11.1

dragonstyle committed:
Update to bslib 0.5.1

dragonstyle committed:
Fix sketchy

cscheid committed:
Render callouts to GFM (#6942)

* format callout for gfm output. Closes #6935
* detect gfm robustly
* changelog

cderv committed:
Handle callout title as empty table

Same fix for epub and reveal than 6721856fab23658df46f69c7f0e0d1fdd3453a7d

cderv committed:
Callout title are no more nil

Same fix as 239a3a9f9aa8df90a85ff7b8b58295245feba9cd

cderv committed:
Also check title correctly for callout attribute

Aranaur committed:
add ukrainian translation

* add _lanhuage-ua.yml with ukrainian translation

Aranaur committed:
add _language-ua.yml to README.md

cderv committed:
Add some more case for html tests of callout

cscheid committed:
fix beamer regression (#6952)

cscheid committed:
typst: turn .block fenced divs into #block calls

cscheid committed:
minimal test

cscheid committed:
Merge pull request #6953 from Aranaur/proposed-fixes

Language .yml file for Ukrainian

cscheid committed:
Merge pull request #6954 from quarto-dev/feature/typst-blocks

Feature: typst blocks

cderv committed:
Add tests for revealjs callout formatting

cderv committed:
Move boostrap update item in NEWS to HTML format part

cderv committed:
Add to changelog revealjs callout fix

[skip ci]

mcanouil committed:
fix: merge css rules into one

mcanouil committed:
fix: merge css rules into one

mcanouil committed:
fix: add mono foreground option

mcanouil committed:
fix: add mono foreground option

mcanouil committed:
chore: merger from main

mcanouil committed:
build: quarto build-js

mcanouil commented:
Closing in favour of:
- #6969
